File: Simple Tic-Tac-Toe/simple_tic_tac_toe.py
n = '         '
game = [[n[0], n[1], n[2]], [n[3], n[4], n[5]], [n[6], n[7], n[8]]]


def print_game():
    print('---------')
    print('|', game[0][0], game[0][1], game[0][2], '|')
    print('|', game[1][0], game[1][1], game[1][2], '|')
    print('|', game[2][0], game[2][1], game[2][2], '|')
    print('---------')


def input_check():
    user_in = input('Enter the coordinates: ')
    while True:
        try:
            x = int(user_in[0])
            y = int(user_in[2])
            break
        except:
            print('You should enter numbers!')
            user_in = input('Enter the coordinates: ')
    a = [x, y]
    return a


def check_coordinates(s):
    user_input = input_check()
    x = int(user_input[0]) - 1
    y = int(user_input[1]) - 1
    while True:
        if x < 0 or x > 2 or y < 0 or y > 2:
            print('Coordinates should be from 1 to 3!')
        elif game[x][y] == 'X' or game[x][y] == 'O':
            print('This cell is occupied! Choose another one!')
        else:
            game[x][y] = s
            break

        user_input = input_check()
        x = int(user_input[0]) - 1
        y = int(user_input[1]) - 1


def play():
    print_game()
    counter = 0
    while counter < 9:
        if counter % 2 == 0:
            check_coordinates('X')
        else:
            check_coordinates('O')

        if game[0][0] == 'X' and game[0][1] == 'X' and game[0][2] == 'X' \
                or game[1][0] == 'X' and game[1][1] == 'X' and game[1][2] == 'X' \
                or game[2][0] == 'X' and game[2][1] == 'X' and game[2][2] == 'X':  # X horizontal
            print_game()
            print('X wins')
            break
        elif game[0][0] == 'O' and game[0][1] == 'O' and game[0][2] == 'O' \
                or game[1][0] == 'O' and game[1][1] == 'O' and game[1][2] == 'O' \
                or game[2][0] == 'O' and game[2][1] == 'O' and game[2][2] == 'O':  # O horizontal
            print_game()
            print('O wins')
            break
        elif game[0][0] == 'X' and game[1][0] == 'X' and game[2][0] == 'X' \
                or game[0][1] == 'X' and game[1][1] == 'X' and game[2][1] == 'X' \
                or game[0][2] == 'X' and game[1][2] == 'X' and game[2][2] == 'X':  # X vertical
            print_game()
            print('X wins')
            break
        elif game[0][0] == 'O' and game[1][0] == 'O' and game[2][0] == 'O' \
                or game[0][1] == 'O' and game[1][1] == 'O' and game[2][1] == 'O' \
                or game[0][2] == 'O' and game[2][1] == 'O' and game[2][2] == 'O':  # O vertical
            print_game()
            print('O wins')
            break
        elif game[0][0] == 'X' and game[1][1] == 'X' and game[2][2] == 'X' \
                or game[0][2] == 'X' and game[1][1] == 'X' and game[2][0] == 'X':  # X diagonal
            print_game()
            print('X wins')
            break
        elif game[0][0] == 'O' and game[1][1] == 'O' and game[2][2] == 'O' \
                or game[0][2] == 'O' and game[1][1] == 'O' and game[2][0] == 'O':  # O diagonal
            print_game()
            print('O wins')
            break
        elif counter > 7:
            print_game()
            print('Draw')
            break
        counter += 1
        print_game()


play()

File: Simple Tic-Tac-Toe/SimpleTicTacToe.java
import java.util.Arrays;
import java.util.Scanner;

public class SimpleTicTacToe {
    public static void main(String[] args) {
        // 1: Create an empty grid array
        String[][] grid = new String[3][3];
        for (String[] strings : grid) {
            Arrays.fill(strings, " ");
        }

        // 2: print the empty grid array
        printTheGrid(grid);

        // 3: play the game
        gameOutcomes(grid);

    }

    // Analyze user input: input should be integer
    public static int[] tryAndCatch() {
        Scanner scan = new Scanner(System.in);
        int x;
        int y;
        while (true) {
            try {
                String i = scan.nextLine();
                x = Integer.parseInt(String.valueOf(i.charAt(0)));
                y = Integer.parseInt(String.valueOf(i.charAt(2)));
                break;
            } catch (Exception e) {
                System.out.println("You should enter numbers!");
            }
            System.out.print("Enter the coordinates: ");
        }
        return new int[]{x, y};
    }

    public static void play(String[][] grid, String s) {
        // 3: Prompt the user to make a move.
        System.out.print("Enter the coordinates: ");

        // 5: check the input
        while (true) {
            int[] arr = tryAndCatch();
            int x = arr[0];
            int y = arr[1];
            --x; --y;
            if (x > 2 || x < 0 || y > 2 || y < 0) {
                System.out.println("Coordinates should be from 1 to 3!");
            } else if (grid[x][y].contains("X") || grid[x][y].contains("O")) {
                System.out.println("This cell is occupied! Choose another one!");
            } else {
                grid[x][y] = s;
                break;
            }
            System.out.print("Enter the coordinates: ");
        }

        // 6: print the grid
        printTheGrid(grid);
    }

    public static void printTheGrid(String[][] grid) {
        for (int i = 0; i < grid.length; i++) {
            if (i == 0) {
                System.out.println("---------");
            }
            for (int j = 0; j < grid[i].length; j++) {
                if (j == 0) {
                    System.out.print("| " + grid[i][j] + " ");
                } else if (j == 2) {
                    System.out.print(grid[i][j] + " |");
                } else {
                    System.out.print(grid[i][j] + " ");
                }
            }
            System.out.println();
            if (i == 2) {
                System.out.println("---------");
            }
        }
    }

    public static void gameOutcomes(String[][] grid) {
        int counter = 0;
        while (counter < 9) {
            if (counter % 2 == 0) {
                play(grid, "X");
            } else {
                play(grid, "O");
            }
            counter++;

            // Game outcomes/conditions
            if (grid[0][0].equals("X") && grid[0][1].equals("X") && grid[0][2].equals("X")
                    || grid[1][0].equals("X") && grid[1][1].equals("X") && grid[1][2].equals("X")
                    || grid[2][0].equals("X") && grid[2][1].equals("X") && grid[2][2].equals("X")) { // X horizontal
                System.out.println("X wins");
                break;
            } else if (grid[0][0].equals("O") && grid[0][1].equals("O") && grid[0][2].equals("O")
                    || grid[1][0].equals("O") && grid[1][1].equals("O") && grid[1][2].equals("O")
                    || grid[2][0].equals("O") && grid[2][1].equals("O") && grid[2][2].equals("O")) { // O horizontal
                System.out.println("O wins");
                break;
            } else if (grid[0][0].equals("X") && grid[1][0].equals("X") && grid[2][0].equals("X")
                    || grid[0][1].equals("X") && grid[1][1].equals("X") && grid[2][1].equals("X")
                    || grid[0][2].equals("X") && grid[1][2].equals("X") && grid[2][2].equals("X")) { // X vertical
                System.out.println("X wins");
                break;
            } else if (grid[0][0].equals("O") && grid[1][0].equals("O") && grid[2][0].equals("O")
                    || grid[0][1].equals("O") && grid[1][1].equals("O") && grid[2][1].equals("O")
                    || grid[0][2].equals("O") && grid[1][2].equals("O") && grid[2][2].equals("O")) { // O vertical
                System.out.println("O wins");
                break;
            } else if (grid[0][0].equals("X") && grid[1][1].equals("X") && grid[2][2].equals("X")
                    || grid[0][2].equals("X") && grid[1][1].equals("X") && grid[2][0].equals("X")) { // X diagonal
                System.out.println("X wins");
                break;
            } else if (grid[0][0].equals("O") && grid[1][1].equals("O") && grid[2][2].equals("O")
                    || grid[0][2].equals("O") && grid[1][1].equals("O") && grid[2][0].equals("O")) { // O diagonal
                System.out.println("O wins");
                break;
            } else if (counter > 6) {
                StringBuilder str = new StringBuilder();
                for (String[] strings : grid) {
                    for (String string : strings) {
                        str.append(string);
                    }
                }
                String s = str.toString();
                if (!s.contains(" ")) System.out.println("Draw");
            }
        }
    }
}
